master/internal/rm/multirm/multirm_intg_test.go

```diff
@@ -0,0 +1,587 @@
+	manager := uuid.NewString()
+	m := &MultiRMRouter{rms: map[string]rm.ResourceManager{manager: &mockRM}}
+
+	res1 := sproto.GetDefaultComputeResourcePoolResponse{PoolName: "default"}
+	res2 := sproto.GetDefaultAuxResourcePoolResponse{PoolName: "default"}
+
+	mockRM.On("GetDefaultComputeResourcePool", manager).Return(res1, nil)
+	mockRM.On("GetDefaultAuxResourcePool", manager).Return(res2, nil)
+
+	actual1, err := m.GetDefaultComputeResourcePool(manager)
+	require.NoError(t, err)
+	require.Equal(t, res1, actual1)
+
+	actual2, err := m.GetDefaultAuxResourcePool(manager)
+	require.NoError(t, err)
+	require.Equal(t, res2, actual2)
+
+	actual1, err = m.GetDefaultComputeResourcePool("bogus")
+	require.Equal(t, err, ErrRPNotDefined("bogus", ""))
+	require.Empty(t, actual1)
+
+	actual2, err = m.GetDefaultAuxResourcePool("bogus")
+	require.Equal(t, err, ErrRPNotDefined("bogus", ""))
+	require.Empty(t, actual2)
+}
+
+func TestValidateResourcePool(t *testing.T) {
+	mockRM := mocks.ResourceManager{}
+	manager := uuid.NewString()
+	m := &MultiRMRouter{rms: map[string]rm.ResourceManager{manager: &mockRM}}
+
+	mockRM.On("ValidateResourcePool", manager, rp).Return(nil)
+
+	err := m.ValidateResourcePool(manager, rp)
+	require.NoError(t, err)
+
+	err = m.ValidateResourcePool("bogus", rp)
+	require.Equal(t, err, ErrRPNotDefined("bogus", rp))
+}
+
+func TestResolveResourcePool(t *testing.T) {
+	mockRM := mocks.ResourceManager{}
+	manager := uuid.NewString()
+	m := &MultiRMRouter{rms: map[string]rm.ResourceManager{manager: &mockRM}}
+
+	req := sproto.ResolveResourcesRequest{ResourcePool: rp}
+
+	mockRM.On("ResolveResourcePool", manager, req).Return(manager, rp, nil)
+
+	resolvedRM, resolvedRP, err := m.ResolveResourcePool(manager, req)
+	require.NoError(t, err)
+	require.Equal(t, manager, resolvedRM)
+	require.Equal(t, rp, resolvedRP)
+
+	resolvedRM, resolvedRP, err = m.ResolveResourcePool("bogus", req)
+	require.Equal(t, err, ErrRPNotDefined("bogus", rp))
+	require.Equal(t, "bogus", resolvedRM)
+	require.Equal(t, rp, resolvedRP)
+}
+
+func TestTaskContainerDefaults(t *testing.T) {
+	mockRM := mocks.ResourceManager{}
+	manager := uuid.NewString()
+	m := &MultiRMRouter{rms: map[string]rm.ResourceManager{manager: &mockRM}}
+
+	res := model.TaskContainerDefaultsConfig{}
+
+	mockRM.On("TaskContainerDefaults", manager, rp, res).Return(res, nil)
+
+	_, err := m.TaskContainerDefaults(manager, rp, res)
+	require.NoError(t, err)
+
+	_, err = m.TaskContainerDefaults("bogus", rp, res)
+	require.Equal(t, err, ErrRPNotDefined("bogus", rp))
+}
+
+func TestGetJobQ(t *testing.T) {
+	mockRM := mocks.ResourceManager{}
+	manager := uuid.NewString()
+	m := &MultiRMRouter{rms: map[string]rm.ResourceManager{manager: &mockRM}}
+
+	res := map[model.JobID]*sproto.RMJobInfo{}
+
+	mockRM.On("GetJobQ", manager, rp).Return(res, nil)
+
+	ret, err := m.GetJobQ(manager, rp)
+	require.NoError(t, err)
+	require.Equal(t, ret, res)
+
+	ret, err = m.GetJobQ("bogus", rp)
```

Comment:
I love how you included the "does not exist" path in all of these; "bogus" is a great way to name it!